src/api/telemetry/TelemetryAPI.js

```diff
@@ -128,6 +133,15 @@ export default class TelemetryAPI {
         return this.subscriptionProviders.filter(supportsDomainObject)[0];
     }
 
+    /**
+     * @private
+     */
+    #findStalenessProvider(domainObject) {
+        return this.stalenessProviders.filter(function (p) {
```

Comment:
You can use `.find()` for this and it will actually be slightly more performant in the worst case because it short-circuits when it finds a match, unlike `filter`.